parking/backend/user_server/wsserver.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from typing import Dict, List
+
+import attr
+from tornado import websocket
+
+import parking.shared.ws_models as models
+from parking.shared.location import Location
+
+
+class UserWSHandler(websocket.WebSocketHandler):
+    """The WebSocket handler. Handles user ws connections."""
+    def check_origin(self, origin) -> bool:
+        return True
+
+    def initialize(self, sessions_handler: 'SessionsHandler') -> None:
+        self.sh = sessions_handler
+
+    def open(self, user_id: str) -> None:
+        self.user_id = user_id
+        logging.info("WebSocket opened for user_id = '{}'".format(self.user_id))
+        self.sh.add_user(user_id, self)
+
+    def on_message(self, message: str) -> None:
+        logging.debug("Received message from user_id = '{}' : '{}'".format(self.user_id, message))
+        msg = models.deserialize_ws_message(message)
+
+        if isinstance(msg, models.LocationUpdateMessage):
+            logging.debug("Received location update from user_id = '{}'".format(self.user_id))
+            self.sh.update_user_location(self.user_id, msg.location)
+        elif isinstance(msg, models.ParkingRequestMessage):
+            logging.debug("Received parking request from user_id = '{}'".format(self.user_id))
+        elif isinstance(msg, models.ParkingAcceptanceMessage):
+            logging.debug("Received parking acceptance from user_id = '{}'".format(self.user_id))
+        elif isinstance(msg, models.ParkingRejectionMessage):
+            logging.debug("Received parking rejection from user_id = '{}'".format(self.user_id))
+            self.sh.add_user_rejection(self.user_id, msg.id)
+        elif isinstance(msg, models.ParkingCancellationMessage):
+            logging.info("Parking cancelled for user_id = '{}'".format(self.user_id))
+            self.close()
+
+    def on_close(self) -> None:
+        self.sh.remove_user(self.user_id)
+        logging.info("WebSocket closed for user_id = {}".format(self.user_id))
+
+
+@attr.s
+class User(object):
+    """Object to represent a connected user."""
+    user_id: str = attr.ib()
+    session: UserWSHandler = attr.ib()
+    location: Location = attr.ib(default=None)
+    rejections: List[int] = attr.ib(default=attr.Factory(list), init=False)
+
+
+class UserSessions(object):
+    """Class to hold references to open ws connections"""
+    def __init__(self) -> None:
+        self.users: Dict[str, User] = {}
+
+    def add_user(self, user_id: str, session: UserWSHandler) -> None:
+        self.users[user_id] = User(user_id, session)
+
+    def remove_user(self, user_id: str) -> User:
+        return self.users.pop(user_id)
```

Comment:
pop takes an index not a value.

This should read `del self.users[user_id]` + a logging message perhaps